### DISCUSSION TITLE
Does Ocelot support conditional routing based on a header?

### DISCUSSION THREAD

**Discussion by usama-azakaw** [ThreeMammals/Ocelot#1998] [Q&A] [answered]
I want to route the request only if there's a bearer token present in the header of the request. (I don't want to validate the token at Ocelot).

**Comment by raman-m** [comment, marked as answer]
You want to detect Auth header but don't want to authenticate, right?
To achieve that you could use any middleware and class which have local `HttpContext` or `HttpRequest` object, mostly all classes and services where you can read [Headers](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20%22.Headers%22&type=code) collection :point_right: 
- use [Middleware Injection](https://ocelot.readthedocs.io/en/latest/features/middlewareinjection.html) and develop a small middleware which detects the header: `context.Request.Headers`, see [HttpRequest.Headers](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httprequest.headers)
- use [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) and develop a small handler to detect the header: `request.Headers`, see [HttpRequestMessage.Headers](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.headers)

After the header detection just fail the request by status code overriding, so use some code from range 3xx-4xx.

Hope it helps!
